lib/oembed/providers.rb

```diff
@@ -205,6 +205,24 @@ def add_official_provider(provider_class, sub_type=nil)
     Instagram << "https://www.instagram.com/p/*"
     add_official_provider(Instagram)
 
+    # Providers for Facebook Posts & Videos
+    # https://developers.facebook.com/docs/plugins/oembed-endpoints
+    FacebookPost = OEmbed::Provider.new('https://www.facebook.com/plugins/post/oembed.json/?url={format}')
```

Comment:
Based on [the Facebook oEmbed documentation](https://developers.facebook.com/docs/plugins/oembed-endpoints), it seems that they only support JSON responses. Furthermore, the `url=` portion of the URL is automatically added to the query string by [the `Provider#build` method](https://github.com/TheCrow1213/ruby-oembed/blob/v0.9.0/lib/oembed/provider.rb#L100-L103).

Please change both the `OEmbed::Provider.new` lines to use the following syntax:
```ruby
FacebookPost = OEmbed::Provider.new('https://www.facebook.com/plugins/post/oembed.json/', :json)
```